### PR TITLE
added RBSM error codes as printouts rather than raw data words

### PR DESCRIPTION
**Pull request by VivaanBahl**

### CODE REVIEW

**Comment by VivaanBahl** [comment]
Has not been tested yet, goal for 2/14/17

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java [hunk 1]
```diff
@@ -216,9 +256,10 @@ public int peel(byte[] buffer, int start, int bytesAvailable) {
             messagePubEnc.publish(estimateVelocity(message.getDataWord()));
         } else if (headerNumber == RBSerialMessage.getHeaderByte("RBSM_MID_ERROR")) {
             // don't want to publish the status ok messages
-            // TODO change this to an enum
             if (message.getDataWord() != 0) {
-                new RobobuggyLogicNotification("RBSM_MID_ERROR:" + message.getDataWord(), RobobuggyMessageLevel.EXCEPTION);
+                RBSMErrorCodes errorCode = RBSMErrorCodes.getErrorCodeForDataWord(message.getDataWord());
```
Comment (abhinavGirish) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/RBSMNode.java]: make sure the lines aren't too long

File: real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        });
+    }
+
+    /**
+     * Runs setup for each test case
+     */
+    @Before
+    public void setUp() {
+        // get a dummy RBSMNode each time
+        testNode = new RBSMNode(NodeChannel.STEERING, NodeChannel.STEERING, null, 100);
```
Comment (abhinavGirish) [inline real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java]: lines too long

File: real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * Tests the RBSM MID error translation from the bit representation to the
+     * error message
+     *
+     * Iterates through each known low-level error and calls peel() with a faked message
```
Comment (abhinavGirish) [inline real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java]: lines a little too long

File: real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            RBSMNode.RBSMErrorCodes errorCode = RBSMNode.RBSMErrorCodes.values()[i];
+            fakeBuffer[4] = (byte) errorCode.getErrorCode();
+            testNode.peel(fakeBuffer, 0, 6);
+
+            waitForMessagePropagation();
+
+            // make sure that we received a message
+            Message top = messages.poll();
+            assertNotNull(top);
+            assertEquals(((RobobuggyLogicNotificationMeasurement) top).getMessage(),
```
Comment (abhinavGirish) [inline real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java]: lines too long here

File: real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    "RBSM_MID_ERROR:" + errorCode.getErrorCode() + ": " + errorCode.getErrorMessage());
+        }
+
+    }
+
+    /**
+     * Unfortunately we can't trigger continuation very easily, so we instead just wait
```
Comment (abhinavGirish) [inline real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/nodes/sensors/RBSMNodeTest.java]: comments too long

**Pull request merged**